src/helper/SM4T_translate.py

```diff
@@ -20,25 +16,59 @@ def check_language_supported_guard(language):
         )
 
 
-def translate_text(text, from_code, to_code):
-    """
-    Translates the given text from the origin language to the target language.
+class Translator:
+    def __init__(self, config: dict):
+        self.device = (
+            f"cuda:{config['device_index']}"
+            if config["translation_device"] == "cuda"
+            else config["translation_device"]
+        )
+        self.tokenizer = SeamlessM4TTokenizer.from_pretrained(
+            config["translation_model"],
+            cache_dir=CONFIG["model_path"],
+        )
+        self.model = SeamlessM4Tv2ForTextToText.from_pretrained(
+            config["translation_model"], cache_dir=CONFIG["model_path"]
+        )
+
+        self.model.config.max_new_tokens = 512
+        self.model.to(self.device)
+
+    def translate_text(self, text, from_code, to_code):
+        """
+        Translates the given text from the origin language to the target language.
 
-    Parameters:
-        text (str): The text to translate.
-        from_code (str): The code of the origin language (e.g., 'en' for English).
-        to_code (str): The code of the target language (e.g., 'de' for German).
+        Parameters:
+            text (str): The text to translate.
+            from_code (str): The code of the origin language (e.g., 'en' for English).
+            to_code (str): The code of the target language (e.g., 'de' for German).
 
-    Returns:
-        str: The translated text.
-    """
-    try:
+        Returns:
+            str: The translated text.
+        """
+        try:
+            text_splitter = RecursiveCharacterTextSplitter(chunk_size=512)
+            text_splitted = text_splitter.split_text(text)
+            translated_chunks = []
+            for chunk in text_splitted:
+                inputs = self.tokenizer(
+                    chunk,
+                    return_tensors="pt",
+                    src_lang=LANGUAGE_MAP[from_code],
+                ).to(self.device)
 
-        inputs = tokenizer(text, return_tensors="pt", src_lang=LANGUAGE_MAP[from_code])
+                with torch.no_grad():
+                    outputs = self.model.generate(
+                        **inputs, tgt_lang=LANGUAGE_MAP[to_code]
+                    )
+                translated_text = self.tokenizer.batch_decode(
+                    outputs, skip_special_tokens=True
```

Comment:
is this right? and what counts as a special token? 
Are for instance chinese characters considered special tokens?

Reply:
No, special tokens are defined by the tokenizer and are like html tags for the tokenizer, <pad> for making the input batches the same size and stuff or <unk> for unknown words